src/main/java/org/broadinstitute/hellbender/engine/samples/PedReader.java

```diff
@@ -0,0 +1,325 @@
+                }
+                individualID = parts[samplePos];
+                if (paternalPos != -1) {
+                    paternalID = maybeMissing(parts[paternalPos]);
+                }
+                if (maternalPos != -1) {
+                    maternalID = maybeMissing(parts[maternalPos]);
+                }
+
+                if (sexPos != -1) {
+                    if (parts[sexPos].equals(SEX_MALE)) {
+                        sex = Gender.MALE;
+                    }
+                    else if (parts[sexPos].equals(SEX_FEMALE)) {
+                        sex = Gender.FEMALE;
+                    }
+                    else {
+                        sex = Gender.UNKNOWN;
+                    }
+                }
+
+                if (phenotypePos != -1) {
+                    if (isQT) {
+                        if (parts[phenotypePos].equals(MISSING_VALUE1)) {
+                            affection = Affection.UNKNOWN;
+                        }
+                        else {
+                            affection = Affection.OTHER;
+                            quantitativePhenotype = parts[phenotypePos];
+                        }
+                    } else {
+                        if (parts[phenotypePos].equals(MISSING_VALUE1)) {
+                            affection = Affection.UNKNOWN;
+                        }
+                        else if (parts[phenotypePos].equals(MISSING_VALUE2)) {
+                            affection = Affection.UNKNOWN;
+                        }
+                        else if (parts[phenotypePos].equals(PHENOTYPE_UNAFFECTED)) {
+                            affection = Affection.UNAFFECTED;
+                        }
+                        else if (parts[phenotypePos].equals(PHENOTYPE_AFFECTED)) {
+                            affection = Affection.AFFECTED;
+                        }
+                        else {
+                            throw new GATKException("Unexpected phenotype type " + parts[phenotypePos] + " at line " + lineNo);
+                        }
+                    }
+                }
+
+                final Sample s = new Sample(individualID, sampleDB, familyID, paternalID, maternalID, sex, affection, quantitativePhenotype);
+                samples.add(s);
+                sampleDB.addSample(s);
+                lineNo++;
+            }
+
+            for (final Sample sample : new ArrayList<Sample>(samples)) {
+                final Sample dad = maybeAddImplicitSample(sampleDB, sample.getPaternalID(), sample.getFamilyID(), Gender.MALE);
+                if (dad != null) {
+                    samples.add(dad);
+                }
+
+                final Sample mom = maybeAddImplicitSample(sampleDB, sample.getMaternalID(), sample.getFamilyID(), Gender.FEMALE);
+                if (mom != null) {
+                    samples.add(mom);
+                }
+            }
+            return samples;
+
+        } catch (IOException e) {
```

Comment:
This catch should probably move up so that it's just around that 1 line of IO code.